shared/common-adapters/icon.native.js

```diff
@@ -116,21 +114,15 @@ class Icon extends React.PureComponent<Props> {
         </Text>
       )
     } else {
-      // We can't pass color to Image, but often we generically pass color to Icon, so instead of leaking this out
-      // let's just filter it out if it exists
-      const imageStyle = has(props.style, 'color') ? omit(props.style, 'color') : props.style
-      icon = <Image source={iconMeta[iconType].require} style={imageStyle} />
+      icon = <Image source={iconMeta[iconType].require} style={props.style} />
```

Comment:
why are we apply style to this and the box?

Reply:
The style was being applied to both before, but with `color` filtered out for the Image, and `color`, `fontSize` and `textAlign` filtered out for the ClickableBox. Those aren't allowed to be style props of the Icon any more.

Reply:
I don't think we want that moving forward. This kind of stuff I think doesn't make sense and likely lead to weird styling being applied

Reply:
Aka. If I add padding I get it twice etc

Reply:
I think we should. My goals are
1. Easy to use component 
2. Sane styling that's typed
3. Not a ton of boilerplate to get it to work correctly (as a consumer of the component)